vars/gdrive.groovy

```diff
@@ -0,0 +1,19 @@
+def upload(String sourceFilePath, String targetFileName, String clientID, String clientSecret, String refreshToken, List<String> parentFolderIDs = [])
+{
+   bat label: "Upload files to Google Drive", script: """
+      SET "AccessToken="
+      FOR /f "tokens=1,2 delims=:, " %%U in ('
+      curl ^
+      --request POST ^
+      --data "client_id=${clientID}&client_secret=${clientSecret}&refresh_token=${refreshToken}&grant_type=refresh_token" ^
+      https://accounts.google.com/o/oauth2/token ^| findstr /i "\"access_token\""
+      ') DO SET "AccessToken=,%%~V"
+      IF DEFINED AccessToken(SET "AccessToken=%AccessToken:~1%") ELSE (SET "AccessToken=n/a")
+      SET AccessToken
+      curl -X POST ^
+      -H "Authorization: Bearer %AccessToken%" ^
+      -F "metadata={ kind: 'drive#file', name: '${targetFileName}', parents: ['${parentFolders.join('\',\'')}'] };type=application/json;charset=UTF-8" ^
```

Comment:
Multiple parent folders?

Reply:
It used to be possible, but it turns out [they removed this as of September 30 this year](https://cloud.google.com/blog/products/g-suite/simplifying-google-drives-folder-structure-and-sharing-models).
The request body still takes a `list` for the property `parents`, though.